[ONNX] More debug logging from fx to onnx

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107654
* #107653
* #107409
* #107408

Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
- Sarif logs will be saved for unittest when `TORCH_LOGS="onnx_diagnostics"` is set.

<img width="640" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/a5718530-3594-46fb-85a2-b8bcc8ba01c7">

BowenBao committed:
[ONNX] More debug logging from fx to onnx

Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.

[ghstack-poisoned]

titaiwangms commented on torch/onnx/_internal/fx/fx_onnx_interpreter.py:
````diff
@@ -520,6 +541,9 @@ def run(
                 )
 
+        with diagnostic.log_section(logging.DEBUG, "ONNX Graph:"):
+            diagnostic.debug("```\n%s\n```", onnxscript_graph.torch_graph)
````
What's the difference to make this one not LazyString?

BowenBao replied:
logging api, adopted from python `logging`, lazy formats the string. Hence in this case as long as getattr `onnxscript_graph.torch_graph` is cheap it is fine. The real formatting happens later.

On the contrary. If let's say your argument is `graph_module.print_readable(print=False)`, which runs codegen when called, you'd need to put it behind `LazyString`.

titaiwangms reviewed (approved):
Not sure how to use the new api. Is there a page that I can learn?

BowenBao committed:
adjust package name; remove xfail for unexpected pass on "[ONNX] More debug logging from fx to onnx"


Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
<img width="640" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/a5718530-3594-46fb-85a2-b8bcc8ba01c7">


[ghstack-poisoned]

BowenBao committed:
use commit from onnxscript main in CI on "[ONNX] More debug logging from fx to onnx"


Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
<img width="640" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/a5718530-3594-46fb-85a2-b8bcc8ba01c7">


[ghstack-poisoned]

BowenBao commented:
> Not sure how to use the new api. Is there a page that I can learn?

The docstrings and notes.

BowenBao committed:
adjust xfail removal between commits; enable sarif log saving for unittest when logs enabled on "[ONNX] More debug logging from fx to onnx"


Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
<img width="640" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/a5718530-3594-46fb-85a2-b8bcc8ba01c7">


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge

BowenBao committed:
remove more xfails on "[ONNX] More debug logging from fx to onnx"


Summary:
- Log fx graph name for 'fx-graph-to-onnx' diagnostic.
- Log fx graph and onnx graph under DEBUG verbosity level for 'fx-graph-to-onnx' diagnostic.
- Adjust unittest to run with diagnostics verbosity level logging.DEBUG.
- Sarif logs will be saved for unittest when `TORCH_LOGS="onnx_diagnostics"` is set.

<img width="640" alt="image" src="https://github.com/pytorch/pytorch/assets/9376104/a5718530-3594-46fb-85a2-b8bcc8ba01c7">


[ghstack-poisoned]

BowenBao commented:
@pytorchbot merge
